asset_proof.js

```diff
@@ -85,18 +109,119 @@ class AssetProof {
     return true;
   }
 
+  /**
+   * @param {AssetProof} other
+   * @return {boolean}
+   */
+  valueEquals(other) {
+    if (!(other instanceof AssetProof)) {
+      return false;
+    }
+
+    return (
+      other.getId() === this.getId() &&
+      other.getAge() === this.getAge() &&
+      other.getNonce() === this.getNonce() &&
+      other.getInput() === this.getInput() &&
+      other.getHash().length === this.getHash().length &&
+      other.getHash().every((e, i) => e === this.getHash()[i]) &&
+      other.getPrevHash().length === this.getPrevHash().length &&
+      other.getPrevHash().every((e, i) => e === this.getPrevHash()[i])
+    );
+  }
+
   /**
    * @return {string}
    */
   toString() {
     return (
       `AssetProof{id=${this.id},` +
-      `age${this.age}},` +
+      `age=${this.age},` +
       `hash=${this.hash},` +
       `nonce=${this.nonce},` +
-      `signature=${this.signature}}`
+      `input=${this.input},` +
+      `hash=${uint8ArrayToBase64(this.hash)},` +
+      `prev_hash=${uint8ArrayToBase64(this.prevHash)},` +
+      `signature=${uint8ArrayToBase64(this.signature)}}`
     );
   }
+
+  /**
+   * @param {Validator} validator
+   * @throws {Error}
+   */
+  validateWith(validator) {
+    const serilized = serialize(
+        this.id,
+        this.age,
+        this.nonce,
+        this.input,
+        this.hash,
+        this.prevHash,
+    );
+
+    if (!validator.validate(serilized, this.signature)) {
+      throw new Error(
+          'The proof signature can\'t be validated with the certificate.',
+      );
+    }
+  }
+}
```

Comment:
equivalent to https://github.com/scalar-labs/scalar/blob/master/common/src/main/java/com/scalar/dl/ledger/proof/AssetProof.java#L140